packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -76,82 +130,199 @@ impl PostgresDatabase {
 
         Ok(())
     }
+
+    /// Inserts a data type into the `property_types` table in the database.
+    async fn insert_property_type(
+        &mut self,
+        version_id: VersionId,
+        property_type: &PropertyType,
+        created_by: AccountId,
+    ) -> Result<(), DatastoreError> {
+        sqlx::query_as::<_, (Uuid,)>(
+            r#"
+            INSERT INTO property_types (
+                version_id,
+                schema,
+                created_by
+            ) 
+            VALUES ($1, $2, $3)
+            RETURNING version_id;
+            "#,
+        )
+        .bind(version_id)
+        .bind(serde_json::to_value(property_type).expect("Invalid property type"))
+        .bind(created_by)
+        .fetch_one(&self.pool)
+        .await
+        .report()
+        .change_context(DatastoreError)
+        .attach_printable("Could not insert property type")?;
```

Comment:
In the long run we probably want these to be variants of the Error right? If so, _perhaps_ we should start doing that now before we build up too much tech debt. If not then I think we should decide when we do want to create more specific Contexts.

Reply:
Addressed in 698eb84bd0e26a1c38df1d7ce1ef199f6985838a